language/types.md

```diff
@@ -124,6 +124,23 @@ foo4: u16 -> u64
 
 -- Works
 bar(foo4)
+
+-- Works for Product Types
+data Smaller:
+    field: i16
+    
+data Bigger:
+    field: i32
+    
+service Service("s"):
+    take_smaller(s: Smaller) -> Smaller
+    take_bigger(b: Bigger) -> Bigger
+    
+func take_smaller(small: Smaller):
+    Service.take_bigger(small)
+    
+func return_bigger(small: Smaller) -> Bigger:
+    <- small
```

Comment:
And here's a counter-examples
```
func take_bigger(big: Bigger):
    Service.take_smaller(big)

func return_smaller(big: Bigger) -> Smaller:
    <- big
```

Just to make it clear what is possible and what isn't